authapi/permissions.py

```diff
@@ -96,7 +116,44 @@ def object_permission_set(self):
         add users to the organization that they are admin for.
         '''
         return Or(
+            AllowOnlySafeHttpMethod,
             AllowAdmin,
             AllowObjectPermission('org:write'),
             AllowObjectPermission('org:admin')
         )
+
+
+TeamCreatePermission = OrganizationUsersPermission
+
+
+class TeamPermission(BaseComposedPermision):
+    '''Permissions for the TeamViewSet.'''
+    def global_permission_set(self):
+        '''All users must be authenticated.'''
+        return AllowOnlyAuthenticated
+
+    def object_permission_set(self):
+        '''
+        admins, users with team:admin for the team, and users with org:admin,
+        or org:write permission for the team's organization have full access
+        to teams. Users with team:read permission for the team, or are a member
+        of the team, or are a member of the team's organization, have read
+        access to the team.
+        '''
+        return Or(
+            AllowAdmin,
+            AllowObjectPermission('team:admin'),
+            AllowObjectPermission('org:admin', 'obj.organization.pk'),
+            AllowObjectPermission('org:write', 'obj.organization.pk'),
+            And(
+                AllowOnlySafeHttpMethod,
+                Or(
+                    AllowObjectPermission('team:read'),
+                    ObjAttrTrue(
+                        'obj.users.filter(pk=request.user.pk).count() >= 1'),
+                    ObjAttrTrue(
+                        'obj.organization.users.filter(pk=request.user.pk)'
+                        '.count() >= 1')
```

Comment:
I think we might be able to do this better using functions instead of `eval()`ing (either by providing a `lambda` as input here that does the same thing, or subclassing `ObjAttrTrue` and overriding a method). Similarly for `AllowObjectPermission`.

Thoughts?